xamarin/Xamarin.Forms
# [Android] Account for user's setting of styles property textAllCaps

### Description of Change ###
Ensure that user's textAllCaps style property setting is accounted for, and not overridden by default TextTransform property

### Issues Resolved ### 
fixes #11703 

### API Changes ###
 None

### Platforms Affected ### 
Android

### Behavioral/Visual Changes ###
When users set textAllCaps style property to false (specifically for a button), the (button) text will no longer appear in the default ALL CAPS

### Before/After Screenshots ### 

**Before:** 
Button text ("Button text") appears in all caps ("BUTTON TEXT"), regardless of whether or not user sets textAllCaps to false

**After:** 
Button text ("Button text") appears in all caps ("BUTTON TEXT"), when user does not set textAllCaps property
<img src="https://user-images.githubusercontent.com/21988533/93616487-c9919180-f9a2-11ea-8d6e-0379f83e8ea3.png" width="500">

Button text ("Button text") appears in expected casing ("Button text"), when user sets textAllCaps to false
<img src="https://user-images.githubusercontent.com/21988533/93616394-b252a400-f9a2-11ea-890a-d65d388aa5e4.png" width="500">

### Testing Procedure ###
This test requires manual review, both on an Android device with API >= 28 and on an Android device with API < 28.
1. In App.cs, switch out `SetMainPage(CreateDefaultMainPage());` for `SetMainPage(Issue11703());`
2. Run on Android, and check that the button text appears in all caps as "BUTTON TEXT". This is the expected default.
3. Now, in styles.xml, comment in the following line: `<item name="android:textAllCaps">false</item>`
4. Run on Android again, and check that the button text no longer appears in all caps, and appears now as "Button text" as expected.
5. Repeat the steps on an Android device with API < 28.

### PR Checklist ###
